### ISSUE TITLE
Strange behavior during solar eclipse

### ISSUE THREAD

**Issue by StarFlea** [Stellarium/stellarium#3706]
Hello folks!

Experiments with the solar eclipse on July 22, 1990 resulted in some strange displays, which I have documented in the screenshots below.

I wanted to find out by experimenting when and where the eclipse could be seen. To do this, I fixed the sun in view and opened the settings for date/time and location.
For example, when I gradually changed the latitude values, the sun suddenly came to be **in front** of the moon:

![1](https://github.com/Stellarium/stellarium/assets/118836358/9fff893b-526f-4f10-b830-d8e343d14812)

In another experiment, I gradually changed the time as the moon moved away from the sun, but the marked lunar landscapes remained with the sun:
![2](https://github.com/Stellarium/stellarium/assets/118836358/2716b157-a958-44ef-8f65-28984cb8e942)

Version: 24.1 Qt5
Windows 10
All graphic drivers are up to date!

Best regards!
StarFlea

[Log.txt](https://github.com/Stellarium/stellarium/files/14874382/Log.txt)

**Comment by alex-w** [comment]
What about atmosphere model?

**Comment by StarFlea** [comment]
Hello

For both issues I completely deactivated the atmosphere and the ground and was able to reproduce both again...

**Comment by alex-w** [comment]
Would you check the behavior when  Stellarium running with `—low-graphics` option?

**Comment by StarFlea** [comment]
Yes, with this option the phemoma also occurs:

![Unbenannt](https://github.com/Stellarium/stellarium/assets/118836358/87a77ba7-7b15-4fdf-8cd6-52f765c83aa4)


Here's the procedure:
1. Select and center the sun
2. Set the date/time, re-center the sun if necessary
3. Hide atmosphere and ground
4. Open location
5. Zoom in to the Sun & Moon
6. Increase / decrease latitude until the moon moves away from the sun's disk: The moon-formation labels remain in place (maybe just their position needs to be updated when the location on Earth changes?)

I could no longer quickly reproduce the behavior of the sun being in front of the moon...

**Comment by StarFlea** [comment]
Ahh, the behaviour only seems to occur, if time rate is paused..

**Comment by gzotti** [comment]
I can finally reproduce it. I had seen this earlier, but thought it was solved. This case is pretty hard to understand. :-(

**Comment by gzotti** [comment]
I also cannot reproduce Sun in front of Moon. However, the labels hopefully are now where they should be.

**Comment by StarFlea** [comment]
OK, now the formation-labels are always with the moon...

But as I tested the fix, the sun comes in front of the moon again:

![2](https://github.com/Stellarium/stellarium/assets/118836358/f470a67e-2fec-47a9-b613-c1d4b34302f6)

I wonder what happened.

All I can say is that I first experimented with the location coordinates and then changed the time by the minute.
I then switched to my Firefox browser window and when I came back to the Stellarium window, the sun covered the moon...

I'm confused...

**Comment by gzotti** [comment]
Aw, yes. Sometimes! Confirmed "strange". :-( I saw it now, one click elsewhere only, and it fixed on the next click.

Re "Sun covers Moon": There is one experimental feature to be configured in the second skylight configure button: "Draw Sun after atmosphere (or similar)". When I did this (for an exhibition), I apparently did not think about eclipses. Make sure you have disabled this option. In any case, when the sun is drawn after atmosphere, it properly covers the Lunar crater labels.
